src/db_structures.php

```diff
@@ -0,0 +1,49 @@
+<?php
+//function that returns the database data
+function data(): PDO
+{
+    $db = new PDO(
+        'mysql:host=DB;dbname=collectorapp',
+        'root',
+        'password'
+
+    );
+    $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+    return $db;
+}
+
+//function that returns only the dog array from database
+function getDogData(PDO $db)
+{
+    $dog_query = $db->prepare("SELECT `id`, `owner_id`, `name`, `breed`, `image` FROM `pet_names` WHERE `species_id` = 2;");
+    $dog_result = $dog_query->execute();
+
+    if ($dog_result) {
+        $dog_data = $dog_query->fetchAll();
+        return $dog_data;
+    } else return 'could not return dog data';
+}
+
+//function that returns only the cat array from database
+function getCatData(PDO $db)
+{
+    $cat_query =$db->prepare("SELECT `id`, `owner_id`, `name`, `breed`, `image` FROM `pet_names` WHERE `species_id` = 1;");
+
+    $cat_result = $cat_query->execute();
+    if ($cat_result){
+        $cat_data = $cat_query -> fetchAll();
+        return $cat_data;
+    } else return 'could not return cat data';
+}
+//function that returns only the owner array from database
+function getOwnerData(PDO $db)
```

Comment:
In general these functions are ambiguously named - unclear if `getOwnerData` only gets data about one owner. You could name it `getOwners` which is clearer that it is going to get all of the owners.

Same with getting dogs and cats.

Name the functions right and you don't need the explanatory comments.